hls4ml/model/hls_layers.py

```diff
@@ -1122,6 +1122,181 @@ def initialize(self):
+        dims = ['N_SEQUENCE_OUT_{}'.format(self.index), 'N_LAYER_{}'.format(self.index)]
+        compression = self.model.config.get_compression(self)
+        reuse_factor = self.model.config.get_reuse_factor(self)
+        self.reuse_factor_recr = reuse_factor
+        if self.model.config.is_resource_strategy(self):
+            if self.model.config.backend.name == 'Vivado':
+                self.model.config.backend.set_closest_reuse_factor(self)
+                self.model.config.backend.set_closest_reuse_factor_recr(self)
+            if compression:
+                self.set_attr('strategy', 'compressed')
+            else:
+                self.set_attr('strategy', 'resource')
+        else:
+            self.set_attr('strategy', 'latency')
+        self.add_output_variable(shape, dims)
+        self.add_weights()
+        self.add_bias()
+        recurrent_weight = self.model.get_weights_data(self.name, 'recurrent_kernel')
+        self.add_weights_variable(name='recurrent_weight', var_name='wr{index}', data=recurrent_weight)
+        recurrent_bias = np.zeros(self.get_output_variable().shape[-1]*4)
+        self.add_weights_variable(name='recurrent_bias', var_name='br{index}', data=recurrent_bias)
+
+
+    def function_cpp(self):
+        params = self._default_function_params()
+        params['w'] = self.get_weights('weight').name
+        params['b'] = self.get_weights('bias').name
+        params['wr'] = self.get_weights('recurrent_weight').name
+        params['br'] = self.get_weights('recurrent_bias').name
+        params['activation'] = self.get_attr('activation')
+        params['recurrent_activation'] = self.get_attr('recurrent_activation')
+
+        return [self._function_template.format(**params)]
+
+    def config_cpp(self):
+        params = self._default_config_params()
+        params['n_in'] = self.get_input_variable().dim_names[1]
+        params['n_sequence'] = self.get_input_variable().dim_names[0]
+        params['n_sequence_out'] = self.attributes['n_sequence_out']
+        params['n_state'] = self.get_output_variable().dim_names[1]
+        params['n_out'] = self.get_output_variable().dim_names[1]
+        params['nzeros'] = self.get_weights('weight').nzeros
+
+        params['config_mult_t1'] = 'config{}_mult1'.format(self.index)
+        params['config_mult_t2'] = 'config{}_mult2'.format(self.index)
+        params['lstm_act_t'] = '{}_config{}_recr'.format(self.get_attr('recurrent_activation'), self.index)
+        params['act_t'] = '{}_config{}'.format(self.get_attr('activation'), self.index)
+        params['strategy'] = self.get_attr('strategy')
+
+        lstm_config = self._config_template[0].format(**params)
+
+
+        lstm_params = self._default_config_params()
+        lstm_params['n_in'] = self.get_output_variable().dim_names[1] + ' * 3'
+        lstm_params['table_size'] = 1024
+        lstm_params['type'] = self.get_attr('recurrent_activation')
+        lstm_params['table_t'] = 'ap_fixed<18,8>'
```

Comment:
`table_size` and `table_t` should be configurable, not hardcoded, especially not to `ap_fixed` string